ml/ad_charts.cc

```diff
@@ -258,6 +258,56 @@ void ml_update_host_and_detection_rate_charts(ml_host_t *host, collected_number
         rrdset_done(host->anomaly_rate_rs);
     }
 
+    /*
+     * Type anomaly rate
+    */
+    {
+        if (!host->type_anomaly_rate_rs) {
+            char id_buf[1024];
+            char name_buf[1024];
+
+            snprintfz(id_buf, 1024, "type_anomaly_rate_on_%s", localhost->machine_guid);
+            snprintfz(name_buf, 1024, "type_anomaly_rate_on_%s", rrdhost_hostname(localhost));
+
+            host->type_anomaly_rate_rs = rrdset_create(
+                    host->rh,
+                    "anomaly_detection", // type
+                    id_buf, // id
+                    name_buf, // name
+                    "anomaly_rate", // family
+                    "anomaly_detection.anomaly_rate", // ctx
```

Comment:
```suggestion
                    "anomaly_detection.type_anomaly_rate", // ctx
```

I think a seperate context makes sense such that ultimatly we would have below contexts:

- anomaly_detection.node_anomaly_rate
- anomaly_detection.type_anomaly_rate
- anomaly_detection.app_anomaly_rate
- anomaly_detection.user_anomaly_rate